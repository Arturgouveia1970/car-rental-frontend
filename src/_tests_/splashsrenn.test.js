import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Splash from '../components/splashScreen/Splash';
import store from '../Redux/configureStore';

describe('Splash screen tests', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Splash />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Splash page has a sign up link', () => {
    const register = render(
      <>
        <Provider store={store}>
          <BrowserRouter>
            <Splash />
          </BrowserRouter>
        </Provider>
      </>,
    );

    expect(register.findByText('Sign up')).toBeTruthy();
  });

  it('Splash page has a log in link', () => {
    const register = render(
      <>
        <Provider store={store}>
          <BrowserRouter>
            <Splash />
          </BrowserRouter>
        </Provider>
      </>,
    );

    expect(register.findByText('Log in')).toBeTruthy();
  });
});
