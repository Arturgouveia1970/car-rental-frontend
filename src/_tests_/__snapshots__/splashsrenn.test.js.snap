// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash screen tests renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="splashbg container-fluid d-flex flex-column align-items-center vw-100"
      >
        <h1
          class="header text-white"
        >
          Dream Cars
        </h1>
        <div
          class="btncontainer"
        >
          <a
            class=""
            href="/login"
          >
            <button
              class="btn btn-primary me-4 fs-4"
              type="button"
            >
              Log in
            </button>
          </a>
          <a
            class=""
            href="/register"
          >
            <button
              class="btn btn-success ms-4 fs-4"
              type="button"
            >
              Sign up
            </button>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="splashbg container-fluid d-flex flex-column align-items-center vw-100"
    >
      <h1
        class="header text-white"
      >
        Dream Cars
      </h1>
      <div
        class="btncontainer"
      >
        <a
          class=""
          href="/login"
        >
          <button
            class="btn btn-primary me-4 fs-4"
            type="button"
          >
            Log in
          </button>
        </a>
        <a
          class=""
          href="/register"
        >
          <button
            class="btn btn-success ms-4 fs-4"
            type="button"
          >
            Sign up
          </button>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
